librad/src/git/storage.rs

```diff
@@ -108,6 +103,24 @@ impl AsRef<git2::Repository> for Storage {
     }
 }
 
+pub struct LinearHistoryCommits<'a>(Option<git2::Commit<'a>>);
+
+impl<'a> Iterator for LinearHistoryCommits<'a> {
+    type Item = git2::Commit<'a>;
+    fn next(&mut self) -> Option<Self::Item> {
+        let next = self
+            .0
+            .as_ref()
+            .and_then(|commit| match commit.parents().len() {
```

Comment:
I would think that it is perfectly fine to be more lenient here: `git2::Revwalk::simplify_first_parent` (equivalent to `git log --first-parent`).

Reply:
My idea is that for entities, where there is history verification, we should not leave room for ambiguity: there should be one and only one parent (or none for revision 1).

Reply:
There is no ambiguity: walking the history backwards along only the first parent will yield exactly the linear history you're after. I don't care much if you use `Revwalk` or a custom iterator that calls `commit.parents().next()`, but if we _require_ that there must be one and only one parent, then I don't understand why we went through all these iterations to end up with three (!) different and redundant linearity requirements: 1. externally hash-linked (git), 2. internally hash-linked, and 3. numeric revision number. Surely one of them must suffice.

Reply:
I eliminated the strictness.
I considered reimplementing it with `Refwalk` keeping the current interface (no `Result`s, just an iterator on `Commit` instances) but the current code seems simpler.